app/scripts/modules/core/src/delivery/details/StageFailureMessage.tsx

```diff
@@ -28,7 +35,15 @@ export class StageFailureMessage extends React.Component<IStageFailureMessagePro
   private getState(props: IStageFailureMessageProps): IStageFailureMessageState {
     const { stage } = props;
     if (stage && (stage.isFailed || stage.isStopped)) {
-      return { isFailed: true, failedTask: (stage.tasks || []).find(t => t.status === 'TERMINAL' || t.status === 'STOPPED') };
+      let failedTask = (stage.tasks || []).find(t => t.status === 'TERMINAL' || t.status === 'STOPPED');
+      let failedStage = stage;
+      let failedStageIndex = -1;
+      if (!failedTask && stage.after) {
+        failedStage = stage.after.find(s => s.status === 'TERMINAL' || s.status === 'STOPPED');
+        failedStageIndex = stage.after.indexOf(failedStage);
+      }
+
+      return { isFailed: true, failedTask: failedTask, failedStage: failedStage, failedStageIndex: failedStageIndex };
```

Comment:
you can simplify this a bit, since the variables and the fields have the same name:
```javascript
return { isFailed: true, failedTask, failedStage, failedStageIndex };
```